src/values.rs

```diff
@@ -67,11 +76,50 @@ impl Type {
         use self::Type::*;
         match *self {
             Duration | TimeStamp | Temperature | ExtNumeric | Color => false,
-            Unit | Bool | String | Json | Binary => true,
+            Unit | String | Json | Binary | OnOff | OpenClosed | ExtBool  => true,
         }
     }
 }
 
+/// An on/off state. Internal representation may be either On or Off.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum OnOff {
+    On,
+    Off,
+}
+
+impl OnOff {
```

Comment:
I agree that they should implement `Eq` and `PartialEq`.

I'm not sure that `as_bool`, `Ord` and `PartialOrd` are very useful, though. I think that we prefer pattern-matching against `On`/`Off` (which offers no ambiguity) to checking whether `as_bool()` is `true`/`false` (which does).

I'd say let's land a first patch without `as_bool`, `PartialOrd` or `Ord` and add them later if we realize that they would be useful. What do you think?